client/iotcore/client_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package iotcore
+
+import (
+	"context"
+	"flag"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/google/uuid"
+	"github.com/mendersoftware/iot-manager/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	model.SetTrustedHostnames([]string{"*.iot.*.amazonaws.com", "localhost"})
+}
+
+var (
+	accessKeyID     string
+	secretAccessKey string
+	endpointURL     string
+)
+
+const testPolicy = `{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Effect": "Allow",
+			"Action": [
+				"iot:Publish",
+				"iot:Receive"
+			],
+			"Resource": [
+				"arn:aws:iot:us-east-1:304194462000:topic/sdk/test/Python"
+			]
+		},
+		{
+			"Effect": "Allow",
+			"Action": [
+				"iot:Subscribe"
+			],
+			"Resource": [
+				"arn:aws:iot:us-east-1:304194462000:topicfilter/sdk/test/Python"
+			]
+		},
+		{
+			"Effect": "Allow",
+			"Action": [
+				"iot:Connect"
+			],
+			"Resource": [
+				"arn:aws:iot:us-east-1:304194462000:client/basicPubSub"
+			]
+		}
+	]
+}`
+
+func init() {
```

Comment:
so in order to run these tests I need to provide the AWS keys, right? CI is taking them from the gitlab settings?

Reply:
Correct; I set up the secrets in Gitlab CI/CD.